c10/core/TensorOptions.h

```diff
@@ -337,7 +337,7 @@ struct C10_API TensorOptions {
 
   // Resolves the tensor type set specified by the current construction axes.
   DispatchKeySet key_set() const noexcept {
-    return DispatchKeySet(computeDispatchKey()).add(DispatchKey::VariableTensorId);
+    return DispatchKeySet(computeDispatchKey()).add(DispatchKey::VariableTensorId).add(DispatchKey::BackendSelectId);
```

Comment:
for my own education: is this the only place where we add default TensorTypeIds?  Seems a bit strange to only be on TensorOptions.

Reply:
Yes, there are: 

https://github.com/pytorch/pytorch/blob/master/c10/core/impl/LocalDispatchKeySet.cpp#L29-L33

https://github.com/pytorch/pytorch/blob/master/c10/core/TensorImpl.cpp#L61